sdk/src/main/java/com/hedera/hashgraph/sdk/Executable.java

```diff
@@ -82,15 +87,30 @@ public SdkRequestT setNodeAccountIds(List<AccountId> nodeAccountIds) {
 
     @FunctionalExecutable
     public CompletableFuture<O> executeAsync(Client client) {
-        return onExecuteAsync(client).thenCompose((v) -> executeAsync(client, 1, null));
+        return onExecuteAsync(client).thenCompose((v) -> {
+            if(nodeAccountIds.isEmpty()) {
+                throw new IllegalStateException("Request node account IDs were not set before executing");
+            }
+            if(nodes.isEmpty()) {
+                for(AccountId id : nodeAccountIds) {
+                    @Nullable
+                    Node node = client.network.networkNodes.get(id);
+                    if(node == null) {
+                        throw new IllegalStateException("Some node IDs did not map to valid nodes in the client network");
+                    }
+                    nodes.add(Objects.requireNonNull(node));
+                }
+            }
+            return executeAsync(client, 1, null);
+        });
     }
 
     private CompletableFuture<O> executeAsync(Client client, int attempt, @Nullable Throwable lastException) {
         if (attempt > maxAttempts) {
             return CompletableFuture.<O>failedFuture(new Exception("Failed to get gRPC response within maximum retry count", lastException));
         }
 
-        var node = client.network.networkNodes.get(getNodeAccountId());
+        var node = nodes.get(nextNodeIndex);
```

Comment:
Not sure if this suggestion is a good one, but since I recommended extracting some stuff from `executeAsync` into `setNodes()` maybe it'd make sense to implement `getNode()` which simply returns `nodes.get(nextNodeIndex)` and is called here `var node = getNode()`?